src/K9OCRS/ClientApp/src/shared/components/Profile.js

```diff
@@ -0,0 +1,322 @@
+    );
+    e.reportValidity();
+  }
+}
+
+async function getUserData(
+  id,
+  setFirst,
+  setLast,
+  setEmail,
+  setRole,
+  setPicture
+) {
+  //Get id from url
+  // const queryParams = new URLSearchParams(window.location.search);
+  // const id = queryParams.get('id');
+  //API call to get user data
+  const inspectedUser = await accountsApi.getUser(id);
+  //Set user data
+  // setUserID(inspectedUser.id);
+  setFirst(inspectedUser.firstName);
+  setLast(inspectedUser.lastName);
+  setEmail(inspectedUser.email);
+  setRole(inspectedUser.userRoleID);
+  setPicture(inspectedUser.profilePictureFilename);
+  //Use to role id to select the radio button
+  const radio = document.getElementById('option' + inspectedUser.userRoleID);
+  radio.checked = true;
+}
+
+function sendPasswordEmail(email, setAlerts) {
+  accountsApi.forgotPassword(email);
+  setAlerts([
+    {
+      color: 'info',
+      message: 'Email for password reset has been sent.',
+    },
+  ]);
+}
+
+function handleSubmit(
+  defaultMode,
+  createMode,
+  inspectMode,
+  setAlerts,
+  userId,
+  first,
+  last,
+  email,
+  password,
+  role,
+  picture
+) {
+  if (defaultMode) {
+    accountsApi.changeInfo(userId, first, last, email);
+    setAlerts([
+      {
+        color: 'success',
+        message: 'Your changes have been saved!',
+      },
+    ]);
+  } else if (createMode) {
+    accountsApi.createUser(email, first, last, password, role);
+    setAlerts([
+      {
+        color: 'success',
+        message: 'User created!',
+      },
+    ]);
+  } else if (inspectMode) {
+    accountsApi.changeInfo(userId, first, last, email, role, picture);
+    setAlerts([
+      {
+        color: 'success',
+        message: 'Your changes have been saved!',
+      },
+    ]);
+  }
+}
+
+const Profile = (props) => {
+  const { currentUser = null, mode = null, paramsId = null } = props;
+  const setAlerts = props.setAlerts;
+  const [userID, setUserID] = useState('');
+  const [first, setFirst] = useState('');
+  const [last, setLast] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [role, setRole] = useState('');
+  const [picture, setPicture] = useState('');
+
+  var defaultMode = false;
+  var createMode = false;
+  var inspectMode = false;
+
+  if (mode === 'create' && currentUser.userRoleID === 1) {
+    createMode = true;
+  } else if (mode === 'inspect' && currentUser.userRoleID === 1) {
+    inspectMode = true;
+  } else {
+    defaultMode = true;
+  }
+
+  useEffect(() => {
+    if (defaultMode) {
+      setUserID(currentUser.id);
+      setFirst(currentUser.firstName);
+      setLast(currentUser.lastName);
+      setEmail(currentUser.email);
+    }
+    if (inspectMode) {
+      setUserID(paramsId);
+      getUserData(paramsId, setFirst, setLast, setEmail, setRole, setPicture);
+    }
+  }, []);
```

Comment:
This is just a band-aid but it shouldn't hurt here as far as I can think rn. ¯\_(ツ)_/¯
```suggestion
  }, []); // eslint-disable-line
```